packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -95,6 +96,7 @@ export default abstract class SqlIntegration
       "segment",
       "rudderstack",
       "ga4",
+      "amplitude",
```

Comment:
I don't love this pattern. I've created a new type `AutoMetricSchemas` that is a union type of the event tracker types that support auto metrics, and I'd love to use that here, but no matter what I've tried, Typescript isn't happy.

With the current pattern, I now have to maintain the list of event trackers that support auto metrics in two places (here, and in the `AutoMetricSchemas` type declaration), which just doesn't feel good for long-term maintenance.

I've tried the following:
- Build an array of supported event trackers here, and derive the `AutoMetricSchemas` type from each index of that array.
- I've tried to define the `AutoMetricSchemas` by extending `SchemaFormatConfig` and using the `Omit` utility. (This returns an error in the `getSchemaFormatConfig` since we don't have a default case there.
- I've tried simply typing the `supportedEventTrackers` array as a `AutoMetricSchemas[]` array, but then the check on line 104 isn't happy.

Open to suggestions

Reply:
I would do your last bullet point.  On line 104, you can make it work with an `as`

```ts
const supportedEventTrackers: AutoMetricSchemas[] = [...];

supportedEventTrackers.includes(this.settings.schemaFormat as AutoMetricSchema)
```

Reply:
This doesn't completely solve the problem since you might add a schema to the type definition, but forget to update this array.  One simple option is to make this an object instead of an array.

```ts
const supportEventTrackers: Record<AutoMetricSchema, true> = {
  segment: true,
  ...
}
```

With this, Typescript will throw an error if you forget to update the code here after adding a new schema to the type.